### PR TITLE
Fix bug 3733@@2

### PR DESCRIPTION
**Pull request by camilesing**
for https://github.com/zstackio/issues/issues/3733

### CODE REVIEW

File: plugin/loadBalancer/src/main/java/org/zstack/network/service/lb/LoadBalancerBase.java [hunk 1]
```diff
@@ -1168,4 +1173,82 @@ private void createListener(final APICreateLoadBalancerListenerMsg msg, final No
...
+            @Override
+            public String getSyncSignature() {
+                return getSyncId();
+            }
+
+            @Override
+            public void run(SyncTaskChain chain) {
+                APIUpdateLoadBalancerListenerEvent evt = new APIUpdateLoadBalancerListenerEvent(msg.getId());
+                LoadBalancerListenerVO lblVo = dbf.findByUuid(msg.getUuid(), LoadBalancerListenerVO.class);
+                if (lblVo == null) {
```
Comment (zxwing) [inline plugin/loadBalancer/src/main/java/org/zstack/network/service/lb/LoadBalancerBase.java]: 改

Comment (camilesing) [inline plugin/loadBalancer/src/main/java/org/zstack/network/service/lb/LoadBalancerBase.java]: chain.next()

File: plugin/loadBalancer/src/main/java/org/zstack/network/service/lb/LoadBalancerApiInterceptor.java [hunk 2]
```diff
@@ -49,8 +66,12 @@ public APIMessage intercept(APIMessage msg) throws ApiMessageInterceptionExcepti
             validate((APIRemoveVmNicFromLoadBalancerMsg) msg);
         } else if (msg instanceof APIGetCandidateVmNicsForLoadBalancerMsg) {
             validate((APIGetCandidateVmNicsForLoadBalancerMsg) msg);
+        } else if(msg instanceof APIUpdateLoadBalancerListenerMsg){
+            validate((APIUpdateLoadBalancerListenerMsg) msg);
         }
 
+        setServiceId(msg);
```
Comment (camilesing) [inline plugin/loadBalancer/src/main/java/org/zstack/network/service/lb/LoadBalancerApiInterceptor.java]: remove

File: plugin/loadBalancer/src/main/java/org/zstack/network/service/lb/LoadBalancerApiInterceptor.java [hunk 3]
```diff
@@ -31,10 +34,24 @@
  */
 public class LoadBalancerApiInterceptor implements ApiMessageInterceptor {
     @Autowired
+    private CloudBus bus;
+    @Autowired
     private DatabaseFacade dbf;
     @Autowired
     private ErrorFacade errf;
 
+    private void setServiceId(APIMessage msg) {
```
Comment (camilesing) [inline plugin/loadBalancer/src/main/java/org/zstack/network/service/lb/LoadBalancerApiInterceptor.java]: remove to -> validate(APIUpdateLoadBalancerListenerMsg msg)

File: plugin/loadBalancer/src/main/java/org/zstack/network/service/lb/LoadBalancerBase.java [hunk 4]
```diff
@@ -1168,4 +1173,82 @@ private void createListener(final APICreateLoadBalancerListenerMsg msg, final No
...
+                    chain.next();
```
Comment (zxwing) [inline plugin/loadBalancer/src/main/java/org/zstack/network/service/lb/LoadBalancerBase.java]: 这个地方调了chain.next()又不调return，后面代码之后不是又要继续执行吗

**Pull request merged**